### PR TITLE
c11 dependency in volk_32u_reverse_32u removed; added .travis.yml

### PR DESCRIPTION
**Pull request by hcab14**
This commit contains two updates
 - fix for compiling without std=c++-11 in volk_32u_reverse_32u
 - travis-ci including cross-compiling for ARM,  see these [logs](https://travis-ci.org/hcab14/volk/builds/344804570)

Let me know if I should split this commit / apply the travis-related changes on top of the work of @n-west.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+language: cpp
+
+dist: trusty
```
Comment (noc0lour) [inline .travis.yml:3]: Is there a newer dist on travis available? trusty is pretty old, xenial would be great

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - env: MATRIX_EVAL="CC=gcc-4.8 && CXX=g++-4.8"
+      addons: {apt: {packages: [*common_packages]}}
+    # Job 2 ... gcc-4.9
+    - env: MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+      addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-4.9]}}
+    # Job 3 ... gcc-5
+    - env: MATRIX_EVAL="CC=gcc-5 && CXX=g++-5"
+      addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-5]}}
+    # Job 4 ... gcc-6
+    - env: MATRIX_EVAL="CC=gcc-6 && CXX=g++-6"
```
Comment (noc0lour) [inline .travis.yml:26]: If you add xenial you can probably add GCC 7 here

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-6]}}
+    # Job 5 ... ARM cross compile (gcc-4.8)
+    - env: MATRIX_EVAL="CC=gcc-4.8 && CXX=g++-4.8 && BOOST_ROOT=$TRAVIS_BUILD_DIR/boost_1_66_0 && CMAKE_ARG=-DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/arm-linux-gnueabihf.cmake"
+      addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-arm-linux-gnueabihf, qemu-user]}}
+      before_script:
```
Comment (noc0lour) [inline .travis.yml]: Also probably add a ARM GCC6  and maybe a GCC7 build?

**Review by noc0lour — COMMENTED** [review]
Looks looks in general.

**Comment by n-west** [comment]
Doesn't travis support multiple distros? Trusty is old, but it's still supported for another year.

**Comment by n-west** [comment]
My travis file looks pretty similar, but since I think you captured everything relevant to trusty and added ARM, I'm happy to just take this

**Comment by noc0lour** [comment]
Oh, I thought trusty was due in 2018, then it's probably OK to keep, but having Xenial in will be a big plus.

**Comment by hcab14** [comment]
> My travis file looks pretty similar, but since I think you captured everything relevant to trusty and added ARM, I'm happy to just take this

I was not aware of your work until @noc0lour pointed me to it.

It would be great to run the tests in lib/volk_test_all one by one instead of all together:
 - it is not necessary to run volk_profile to see which test has failed
 - running volk_profile complicates .travis.yml for ARM

> Oh, I thought trusty was due in 2018, then it's probably OK to keep, but having Xenial in will be a big plus.

According to
https://docs.travis-ci.com/user/reference/overview/#Virtualization-environments
Ubuntu trusty is the most recent linux virtualization environment, but GCC up to gcc-7 is made available.

**Comment by n-west** [comment]
Yep, to use newer versions you have to use docker (https://github.com/travis-ci/travis-ci/issues/5821), but I won't require that to merge this. Do you have public results running somewhere?

**Comment by n-west** [comment]
nevermind, trying it out on mine: https://travis-ci.org/n-west/volk/builds/344843570

**Comment by hcab14** [comment]
Can you access this log https://travis-ci.org/hcab14/volk/builds/344834078 ?

**Comment by n-west** [comment]
Yea, I looked at it after I merged the readme stuff. I'm going to get permission for travis to access this repo (gnuradio/volk) and change the readme marker thing

**Pull request merged**